src/commands/utils/udfRegistration.test.ts

```diff
@@ -171,4 +184,247 @@ describe("commands/utils/udfRegistration", () => {
       assert.deepStrictEqual(result, []);
     });
   });
+
+  describe("registerMultipleUdfs", () => {
+    const exampleDatabase = { provider: "aws", region: "us-east-1" } as any;
+    function makeRegistration(name: string): UdfRegistrationData {
+      return {
+        classInfo: { className: `com.acme.${name}Class`, simpleName: `${name}Class` },
+        functionName: name,
+      };
+    }
+    let withProgressStub: sinon.SinonStub;
+    let getDbStub: sinon.SinonStub;
+    let loaderStub: sinon.SinonStub;
+    let infoNotifStub: sinon.SinonStub;
+    let errorMsgStub: sinon.SinonStub;
+    let fireStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      withProgressStub = sandbox
+        .stub(vscode.window, "withProgress")
+        .callsFake(async (_o: any, task: any) => task({ report: () => {} }));
+      getDbStub = sandbox
+        .stub(FlinkDatabaseViewProvider, "getInstance")
+        .returns({ database: exampleDatabase } as unknown as FlinkDatabaseViewProvider);
+      loaderStub = sandbox.stub(CCloudResourceLoader, "getInstance").returns({
+        executeBackgroundFlinkStatement: sandbox.stub().resolves(undefined),
+      } as unknown as CCloudResourceLoader);
+      infoNotifStub = sandbox.stub(notifications, "showInfoNotificationWithButtons").resolves();
+      errorMsgStub = sandbox.stub(vscode.window, "showErrorMessage").resolves(undefined);
+      fireStub = sandbox.stub(emitters.udfsChanged, "fire");
+    });
+
+    it("throws when no database selected", async () => {
+      getDbStub.returns({ database: undefined } as unknown as FlinkDatabaseViewProvider);
```

Comment:
Then with a stubbed provider instance:
```suggestion
      stubbedProvider.database.get(() => undefined);
```